zipline/assets/asset_writer.py

```diff
@@ -165,6 +165,62 @@ def _generate_output_dataframe(data_subset, defaults):
     return output
 
 
+# Define a version number for the database generated by these writers
+# Increment this version number any time a breaking change is made to the
+# schema and readers of the database
+ASSET_DB_VERSION = 1
+
+
+def check_version_info(version_table, expected_version, write_version=False):
+    """
+    Checks for a version value in the version table, and inserts version
+    info if none is present.
+
+    Parameters
+    ----------
+    version_table : sa.Table
+        The version table of the asset database
+    expected_version : int
+        The expected version of the asset database
+    write_version : bool, optional
+        If True then the current version will be written in to the
+        version_table if no version information is present
+
+    Raises
+    ------
+    AssetsDbVersionError
+        If the version is in the table and not equal to ASSETS_DB_VERSION.
+    """
+
+    # Read the version out of the table
+    table_version = sa.select((version_table.c.version,)).scalar()
```

Comment:
The table will have a single row or no rows. Do you know of any way to enforce that on the table?

It means the version in the table - I'll rename it to 'version_in_table' or something more clear.